specification/programmableconnectivity/Azure.ProgrammableConnectivity/apis/network.tsp

```diff
@@ -0,0 +1,58 @@
+import "./common.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+
+namespace Azure.ProgrammableConnectivity;
+
+/// Interfaces
+
+interface Networks {
+  @doc("Retrieves the network a given device is on. Returns network in a networkCode format that can be used for other APIs.")
+  retrieve is Operations.ResourceAction<
```

Comment:
**Question:** One of Dominik's suggestions on your word document was to implement Networks as a resource that we can GET, rather than using a POST retrieve. 

My instinct is that what you've got here is correct, but it would be good to see an explanation (either here or in the word doc) of why we're doing it this way

Reply:
Partly I didn't think about it very much once I decided it was better to stick with what we have than do the headers. I had three reasons for not following this pattern though I think: 
1) This is an API that we might deprecate before GA so I don't know whether its worth putting much effort into it at this point. 
2) Consistency across our APIs feels valuable 
3) This is potentially an error of judgement on my part but I think viewing networks as the resource with IP as the key felt odd to me, I think becuase I think of the network an IP is on as being a feature of the IP address not the network. It feels like an extension of the "retrieving some information about a device" pattern that we have for the other APIs. 

But essentially- I think this way is fine and I think editing this API to be potentially slightly better is very low priority at this point. I'll put it in the excel sheet as something to consider though!

Reply:
I think that you're right - point 3 is the most important one for me. We don't maintain state about networks, and we don't expect users to "get" them using a known ID, which is what a RESTful design here would imply - as you say, this API is like the others, in that it's a function that accepts a parameter, so it makes sense for it to be a POST as well. It'd be good if you could add this thinking to the document so we record that we've considered that comment.